flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java

```diff
@@ -677,6 +679,31 @@ public InlineElement getDescription() {
                                             code(SchedulerType.AdaptiveBatch.name()))
                                     .build());
 
+    @Experimental
+    @Documentation.Section({
+        Documentation.Sections.EXPERT_SCHEDULING,
+        Documentation.Sections.ALL_JOB_MANAGER
+    })
+    public static final ConfigOption<Boolean> SCHEDULER_PREFER_MINIMAL_TASKMANAGERS_ENABLED =
+            key("jobmanager.adaptive-scheduler.prefer-minimal-taskmanagers")
+                    .booleanType()
+                    .defaultValue(true)
```

Comment:
```suggestion
                    .defaultValue(false)
```
Shouldn't we have the feature disabled by default to have a more stable upgrade? Users can enable it if they need to explicitly.

Reply:
After the mailing list and PR conversation IMO the consensus is that minimizing TM number is desired in most use-cases, so I am leaning towards changing this option even though it's a patch release, and include some warning/details about it in the Release Notes.

I would say most people do not read docs only if something do not work as they think it should, so I would prefer making it work as one would expect by default the most as we can. But I do not necessarily insist, if we decide to opt this out, we can still highlight to opt-in in the release notes, but back to base, most people won't read that anyways. :)

Reply:
> After the mailing list and PR conversation IMO the consensus is that minimizing TM number is desired in most use-cases, so I am leaning towards changing this option even though it's a patch release, and include some warning/details about it in the Release Notes.
> 
> I would say most people do not read docs only if something do not work as they think it should, so I would prefer making it work as one would expect by default the most as we can. But I do not necessarily insist, if we decide to opt this out, we can still highlight to opt-in in the release notes, but back to base, most people won't read that anyways. :)

So, if it doesn't work as expected (i.e. that too many TMs being used) they should be eventually guided to FLINK-33977, this PR or the configuration parameter. And we could also highlight that in the release notes.

But I also see your point that the ML discussion considered it a "some-what bug"/unexpected behavior.

For me, introducing this feature with the feature flag/config parameter being disabled is still the best compromise between version stability and fixing this bug.

Reply:
That said, I won't block my approval since it looks like the everyone else tends to prefer the approach with the feature flag being enabled based on the thumbs up in the [other comment](https://github.com/apache/flink/pull/25218#issuecomment-2872819736).